packages/react-ui-core/src/Ratings/RatingBar.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import RatingItem from './RatingItem'
+
+export default class RatingBar extends Component {
+  static propTypes = {
+    theme: PropTypes.object,
+    ratingScale: PropTypes.number,
+    rating: PropTypes.number,
+    backgroundColor: PropTypes.string,
+    total: PropTypes.string,
+    stroke: PropTypes.string,
+    strokeWidth: PropTypes.number,
+    itemHeight: PropTypes.string,
+    itemWidth: PropTypes.string,
+    pathD: PropTypes.string.isRequired,
+    viewBox: PropTypes.string.isRequired,
+  }
+  static defaultProps = {
+    theme: {},
+  }
+
+  render() {
+    const {
+      ratingScale,
+      rating,
+      backgroundColor,
+      total,
+      theme,
+      stroke,
+      strokeWidth,
+      itemHeight,
+      itemWidth,
+      pathD,
+      viewBox,
+      polygonPoints,
+    } = this.props
+    const ratingItems = []
+    let currentRating = rating
+    while (ratingItems.length < ratingScale) {
+      const ratingValue = currentRating - 1 > 0 ?
+        1 : currentRating
+      currentRating -= 1
+      const val = ratingValue < 0 ? 0 : ratingValue
+      ratingItems.push(
```

Comment:
Please try to use newlines to create better separation between code.  This is currently a nightmare to try to read.